src/test/jetpack-editor-setup.js

```diff
@@ -47,15 +47,26 @@ const jetpackEmbedVariations = [
 	'smartframe',
 ];
 
+/**
+ * Registers the Jetpack blocks using a sandbox registry for the modules via `jest.isolateModules`.
+ * Module isolation is required in this case because the Jetpack blocks are registered upon module importation.
+ *
+ * @param {Object} props
+ * @return {Object} Blocks API registered in the sandbox registry.
+ */
+const registerJetpackBlocksIsolated = ( props ) => {
+	let blocksAPI;
+	jest.isolateModules( () => {
+		registerJetpackBlocks( props );
+		blocksAPI = require( '@wordpress/blocks' );
+	} );
+	return blocksAPI;
```

Comment:
We need to return the blocks API because a new instance of the block registry is created when isolating modules.